src/assets/CollectibleDetectionController.test.ts

```diff
@@ -328,50 +346,79 @@ describe('CollectibleDetectionController', () => {
   });
 
   it('should not autodetect collectibles that exist in the ignoreList', async () => {
+    const selectedAddress = '0x2';
     collectibleDetection.configure({
       networkType: MAINNET,
       selectedAddress: '0x2',
     });
+    collectiblesController.configure({ selectedAddress });
+
+    const { chainId } = collectibleDetection.config;
+
     await collectibleDetection.detectCollectibles();
-    expect(collectiblesController.state.collectibles).toHaveLength(1);
+    expect(
+      collectiblesController.state.allCollectibles[selectedAddress][chainId],
+    ).toHaveLength(1);
     expect(collectiblesController.state.ignoredCollectibles).toHaveLength(0);
     collectiblesController.removeAndIgnoreCollectible(
-      '0x1d963688fe2209a98db35c67a041524822cf04ff',
+      '0x1d963688FE2209A98dB35C67A041524822Cf04ff',
       '2577',
     );
-    await collectibleDetection.detectCollectibles();
-    expect(collectiblesController.state.collectibles).toHaveLength(0);
+
     expect(collectiblesController.state.ignoredCollectibles).toHaveLength(1);
+    await collectibleDetection.detectCollectibles();
+    expect(
+      collectiblesController.state.allCollectibles[selectedAddress][chainId],
+    ).toHaveLength(0);
   });
 
   it('should not detect and add collectibles if there is no selectedAddress', async () => {
-    collectibleDetection.configure({ networkType: MAINNET });
+    const selectedAddress = '';
+    collectibleDetection.configure({
+      networkType: MAINNET,
+      selectedAddress,
+    });
+    const { chainId } = collectibleDetection.config;
     await collectibleDetection.detectCollectibles();
-    expect(collectiblesController.state.collectibles).toStrictEqual([]);
+    const { allCollectibles } = collectiblesController.state;
+    expect(allCollectibles[selectedAddress]?.[chainId] || []).toStrictEqual([]);
   });
 
   it('should not detect and add collectibles to the wrong selectedAddress', async () => {
     collectibleDetection.configure({
       networkType: MAINNET,
       selectedAddress: '0x9',
     });
+    const { chainId } = collectibleDetection.config;
+
     collectiblesController.configure({ selectedAddress: '0x9' });
     collectibleDetection.detectCollectibles();
     collectibleDetection.configure({ selectedAddress: '0x12' });
     collectiblesController.configure({ selectedAddress: '0x12' });
     await new Promise((res) => setTimeout(() => res(true), 1000));
     expect(collectibleDetection.config.selectedAddress).toStrictEqual('0x12');
-    expect(collectiblesController.state.collectibles).toStrictEqual([]);
+
+    expect(
+      collectiblesController.state.allCollectibles[
+        collectibleDetection.config.selectedAddress
+      ]?.[chainId] || [],
+    ).toStrictEqual([]);
```

Comment:
[nit] ```suggestion
const { chainId, selectedAddress } = collectibleDetection.config;

    expect(
      collectiblesController.state.allCollectibles[selectedAddress]?.[chainId],
    ).toStrictEqual([]);
```
why was `|| []` in there? seems like it undermines the test?

Reply:
Here's the reason: previous tests were against these 'calculated properties', these were always an array, that's why the tests that are expecting emptiness are using `toStrictEqual([])`.

Emptiness in the state now is represented by the absence of the key (address or chainId), for example if you switch to `selectedAddress` `'0x2'` and `chainId` `'56'`, the data structure will not have that nested object, ie: `collectiblesController.state.allCollectibles['0x2']['56']` will throw because `collectiblesController.state.allCollectibles['0x2']` will be undefined, since there are no collectibles.

The `|| []` is used as fallback to match the previous tests, this is also used in the state selectors on mobile, because components props are expecting an array. (https://github.com/MetaMask/metamask-mobile/pull/3458/files#diff-065d8fc10fb71f841b46676cb75d0242d8ec471c6a912ce1142e4ba121fee5c1R12-R30)

We could definitely change the expectation to be `undefined` or `[]`  in according on each test case.

Reply:
Ok yeah I think it makes more sense to change the expectation to `undefined` because that is actually what we expect right? Rather than changing it to an empty array in the test itself if `undefined`.

Reply:
And in many of these cases you've included the `|| []` we don't even expect them to be `undefined`

Reply:
Right, there are some tests that add, expect, remove, expect, in that case the array existed and now it should be `[]` instead of `undefined`. But I'll update them that way.